spinalcordtoolbox/scripts/sct_image.py

```diff
@@ -517,36 +516,28 @@ def multicomponent_merge(im_in_list: Sequence[Image]):
 
 
 def visualize_warp(im_warp: Image, im_grid: Image = None, step=3, rm_tmp=True):
-    fname_warp = im_warp.absolutepath()
+    fname_warp = im_warp.absolutepath
     if im_grid:
-        fname_grid = im_grid.absolutepath()
+        fname_grid = im_grid.absolutepath
     else:
-        from numpy import zeros
         tmp_dir = tmp_create()
-        status, out = run_proc(['fslhd', fname_warp])
+        nx, ny, nz = im_warp.data.shape[0:3]
         curdir = os.getcwd()
         os.chdir(tmp_dir)
-        dim1 = 'dim1           '
-        dim2 = 'dim2           '
-        dim3 = 'dim3           '
-        nx = int(out[out.find(dim1):][len(dim1):out[out.find(dim1):].find('\n')])
-        ny = int(out[out.find(dim2):][len(dim2):out[out.find(dim2):].find('\n')])
-        nz = int(out[out.find(dim3):][len(dim3):out[out.find(dim3):].find('\n')])
-        sq = zeros((step, step))
+        sq = np.zeros((step, step))
         sq[step - 1] = 1
         sq[:, step - 1] = 1
-        dat = zeros((nx, ny, nz))
+        dat = np.zeros((nx, ny, nz))
         for i in range(0, dat.shape[0], step):
             for j in range(0, dat.shape[1], step):
                 for k in range(dat.shape[2]):
                     if dat[i:i + step, j:j + step, k].shape == (step, step):
                         dat[i:i + step, j:j + step, k] = sq
-        fname_grid = 'grid_' + str(step) + '.nii.gz'
         im_grid = Image(param=dat)
         grid_hdr = im_warp.hdr
         im_grid.hdr = grid_hdr
-        im_grid.absolutepath = fname_grid
-        im_grid.save()
+        fname_grid = 'grid_' + str(step) + '.nii.gz'
+        im_grid.save(fname_grid)
         fname_grid_resample = add_suffix(fname_grid, '_resample')
         run_proc(['sct_resample', '-i', fname_grid, '-f', '3x3x1', '-x', 'nn', '-o', fname_grid_resample])
```

Comment:
Since this is already a "various fixes" PR, it might be a nice and easy time to replace the `run_proc` calls in this function with `main` calls. 

(There's another `run_proc` call for `sct_apply_transfo` a few lines down, too.)

Reply:
Fixed in ecad436.